src/test/controllerscriptenginelegacy_test.cpp

```diff
@@ -693,13 +685,8 @@ TEST_F(ControllerScriptEngineLegacyTest, screenWontSentRawDataIfNotConfigured) {
             "Could not find a valid transform function but the screen doesn't "
             "accept raw data. Aborting screen rendering.");
 
-    transformScreenFrameFunctions().insert(
-            dummyScreen.identifier,
-            newTransformScreenFrameFunction(
-                    QMetaMethod(),
-                    false));
     renderingScreens().insert(dummyScreen.identifier, pDummyRender);
-    rootItems().insert(dummyScreen.identifier, std::make_shared<QQuickItem>());
+    rootItems().insert(dummyScreen.identifier, new mixxx::qml::QmlMixxxController());
```

Comment:
this leaks. Instead consider inserting the value directly (instead storing a pointer). You can still take the address of the object later.

Reply:
I've added a free in the teardown. This pointer is owned by the QML scene. I have added a comment in the header file.